sound/soc/intel/boards/sof_nau8825.c

```diff
@@ -478,17 +424,13 @@ sof_card_dai_links_create(struct device *dev, enum sof_ssp_codec amp_type,
 			links[id].ops = &max_98373_ops;
 			break;
 		case CODEC_NAU8318:
-			links[id].codecs = nau8318_components;
-			links[id].num_codecs = ARRAY_SIZE(nau8318_components);
-			links[id].init = speaker_codec_init;
+			nau8318_set_dai_link(&links[id]);
 			break;
 		case CODEC_RT1015P:
 			sof_rt1015p_dai_link(&links[id]);
 			break;
 		case CODEC_RT1019P:
-			links[id].codecs = rt1019p_component;
-			links[id].num_codecs = ARRAY_SIZE(rt1019p_component);
-			links[id].init = speaker_codec_init;
+			sof_rt1019p_dai_link(&links[id]);
```

Comment:
.. and this belong in a different patch IMHO, that's not related to Nuvoton and that's clearly a cleanup we missed earlier.

Reply:
done